pallets/ddc-verification/src/lib.rs

```diff
@@ -2430,6 +2553,54 @@ pub mod pallet {
 			}
 		}
 
+		pub(crate) fn get_consensus_for_sub_aggregates(
+			cluster_id: &ClusterId,
+			era_id: DdcEra,
+			activities: Vec<SubAggregates>,
+			min_nodes: u16,
+			threshold: Percent,
+		) -> Result<(Vec<SubAggregates>, Vec<SubAggregates>), Vec<OCWError>> {
+			let mut customer_buckets: BTreeMap<ActivityHash, Vec<SubAggregates>> = BTreeMap::new();
+
+			// Flatten and collect all customer activities
+			for activity in activities.iter() {
+				customer_buckets
+					.entry(activity.get_consensus_id::<T>())
+					.or_default()
+					.push(activity.clone());
+			}
+
+			let mut consensus_activities = Vec::new();
+			let mut not_consensus_activities = Vec::new();
+			let mut errors = Vec::new();
+			let min_threshold = threshold * min_nodes;
+
+			// Check if each customer/bucket appears in at least `min_nodes` nodes
+			for (id, activities) in customer_buckets {
+				if activities.len() < min_nodes.into() {
+					not_consensus_activities.extend(activities);
+				} else if let Some(activity) =
+					Self::reach_consensus(&activities, min_threshold.into())
+				{
+					consensus_activities.push(activity);
+				} else {
+					errors.push(OCWError::ActivityNotInConsensus {
```

Comment:
id actually is an activity hash. i think it's not useful to have only activity hash in an error. Can you get aggregate key (either bucket id or node pub key) from it?

imagine that you are collecting aggregates for bucket A and you got 3 different aggregates from 3 aggregators. so you have 3 different activity hashes, and you have 3 errors that contain era id, cluster id and activity hash.

as for me it's not correct. in case you don't have a consensus (there are different activity hashes for the same aggregate key), it should be 1 error from which you can understand what aggregate it is, and who provided what hashes (if needed)